File: OptiPlanBackend/OptiPlanBackend/Models/ProjectTask.cs

using OptiPlanBackend.Enums;
using System.ComponentModel.DataAnnotations;

namespace OptiPlanBackend.Models
{
    public class ProjectTask
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;

        [StringLength(1000)]
        public string Description { get; set; } = string.Empty;

        public Guid ProjectId { get; set; }
        public Project Project { get; set; }

        public Guid? AssignedUserId { get; set; }
        public User? AssignedUser { get; set; }

        public Guid? ReporterId { get; set; }
        public User? Reporter { get; set; }

        public OptiPlanBackend.Enums.TaskStatus Status { get; set; } = OptiPlanBackend.Enums.TaskStatus.ToDo;
        public TaskPriority Priority { get; set; } = TaskPriority.Medium;
        public TaskType Type { get; set; } = TaskType.Task;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DueDate { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? CompletedAt { get; set; }

        public int EstimatedHours { get; set; }
        public int? StoryPoints { get; set; }
        public double CompletionPercentage { get; set; } = 0;

        public string? Labels { get; set; }
        public bool IsBlocked { get; set; } = false;
        public string? BlockReason { get; set; }

        // ✅ Sprint relationship
        public Guid? SprintId { get; set; }
        public Sprint? Sprint { get; set; }

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
        public ICollection<TaskHistory> History { get; set; } = new List<TaskHistory>();

        public bool IsCompleted { get; internal set; }
    }
}

File: optiplan-frontend/src/app/models/projectTask.ts
import { User } from "./user";

export interface ProjectTask {
  id: string;  // Changed from number to string to match Guid
  title: string;  // Changed from Name to Title
  description: string;
  status: 'ToDo' | 'InProgress' | 'InReview' | 'Done' | 'Rejected' | 'Archived';
  type: 'Task' | 'Bug' | 'Feature' | 'Improvement' | 'Epic';  
  priority: 'Low' | 'Medium' | 'High' | 'Critical'; 
  projectId: string;
  AssignedUser?: User;
  reporter?: User;
  createdAt: Date;
  updatedAt: Date;
  dueDate?: Date;
  startDate?: Date;
  completedAt?: Date;
  estimatedHours: number;
  storyPoints?: number;
  completionPercentage: number;
  labels?: string;
  sprintId?: string;
  isBlocked: boolean;
  blockReason?: string;
  isCompleted: boolean;
}